src/script/sign.cpp

```diff
@@ -202,7 +202,7 @@ bool ProduceSignature(const BaseSignatureCreator& creator, const CScript& fromPu
     uint160 scriptAddress;
     sigdata.scriptWitness.stack.clear();
 
-    if (solved && whichType == TX_SCRIPTHASH || whichType == TX_PARAMETERIZED_SCRIPTHASH)
+    if ((solved && whichType == TX_SCRIPTHASH) || whichType == TX_PARAMETERIZED_SCRIPTHASH)
```

Comment:
should be

if( solved && (whichType == TX_SCRIPTHASH || whichType == TX_PARAMETERIZED_SCRIPTHASH))